vault/logfile_api.php

```diff
@@ -0,0 +1,23 @@
+<?php 
+	require 'classes/LogFileViewer.php';
+	// mocking cidram array for now
+	$CIDRAM = array();
+	$CIDRAM['Vault'] = __DIR__."/";
+	$CIDRAM['Config'] = parse_ini_file('config.ini', true);
+
+	if (isset($_REQUEST['buffer_pos']) && isset($_REQUEST['category'])) {
+
+			$category = $_REQUEST['category'];
+
+			$buffer_pos = $_REQUEST['buffer_pos'];
+
+			$logFileViewer  = new LogFileViewer($CIDRAM);
+
+			$response  = $logFileViewer->getResponse($buffer_pos, $category);
+
+			echo json_encode($response);
```

Comment:
Something else I thought about just now, after having already submitted my review here: Creating a new, dedicated repository might be better than using the extras repository, for the reason that the extras repository doesn't currently have any supported for tests (there less complications when creating a new, dedicated repository, because we could make sure that has support for tests, independently of any concern or consideration for generally unrelated code).

Reply:
Like Maikuolan said, modularity. And my vote is on a separate repository because as far as I know our user base not everyone will want this. But it is a good extra feature / separate addition.